### ISSUE TITLE
Notification Popup URL Issue

### ISSUE THREAD

**Issue by jasonkuehl** [raelgc/scudcloud#308]
![selection_008](https://cloud.githubusercontent.com/assets/9578513/11570500/8b9036f8-99c5-11e5-8cef-62886349f0bd.png)

When a URL is sent that is to long it will leave off the ending add a ... at the end in the Notification message. Once it does that if you can click on the URL in the notification it will leave of the last part of the URL that was behind the ...

![selection_009](https://cloud.githubusercontent.com/assets/9578513/11570543/bd145402-99c5-11e5-9f73-ae7cd6de2c70.png)

I'm explaining this horrible. So I attached pictures.

**Comment by PaulFurtado** [comment]
I also have this problem, I was just coming here to file an issue, funny that someone made one just 5 hours before me.

**Comment by raelgc** [comment]
Hi @jasonkuehl and @PaulFurtado, thanks for reporting this issue.

Sorry, can you confirm the problem? When you click in the notification, the link opened in browser is the shortened one with the appended `...`?

So, something like `http://www.example.com?abc=def` (if it was too long, of course) will turn `http://example.com?abc=...`, is that?

**Comment by jasonkuehl** [comment]
Yes, correct. @raelgc . When you click on the notification it will open a web browser to http://example.com?abc=    not the intended http://www.example.com?abc=def 

@PaulFurtado You're also in Boston, ,maybe its a location issue :P

**Comment by raelgc** [comment]
What desktop environment are you guys using?

**Comment by jasonkuehl** [comment]
@raelgc Ubuntu 15.04 with Cinnamon

**Comment by PaulFurtado** [comment]
@raelgc gnome-shell here

**Comment by raelgc** [comment]
This issue is a problem... because the notification system is turning the URL in a hyperlink, not ScudCloud.

**Comment by PaulFurtado** [comment]
@raelgc thanks for looking into it. Other messaging apps do get around this issue (the URL will appear truncated, but the hyperlink remains intact), so there must be a hack. Is scudcloud truncating the text or is it the notification system?

**Comment by raelgc** [comment]
ScudCloud is getting the Slack notification directly and sending to system. So, or Slack is truncating or notification system. Probably if we add a simple `print` to `notify.py` we'll discover.

**Comment by jasonkuehl** [comment]
Interesting, I did know that’s how SC worked even from looking over the code. I think the Print idea would work however.

**Comment by raelgc** [comment]
Can you guys tweak `notify.py` and add a `print` of the notification, then paste it here, please?

**Comment by jasonkuehl** [comment]
I only see notifier.py  and notify2.py 

[15991:2253 0:3033] 03:16:33 Tue Feb 16 [jkuehl@TheDestroyer: +1] /usr/lib/python3/dist-packages/scudcloud 
(1:3033)$ cat notif
notifier.py  
notify2.py

**Comment by raelgc** [comment]
@jasonkuehl My bad: `notifier.py`.

**Comment by raelgc** [comment]
I just published a new version (`1.16`) where if you start with `scudcloud --debug=True` you'll be able to see the notification printed in the command line.

**Comment by jasonkuehl** [comment]
Notification: title [New message from GUY] message [what-if.xkcd.com/...] icon [/tmp/scudcloud_U08U4Q96C.png]

(1:3173)$ scudcloud --version
ScudCloud 1.16
Python 3.4.3
Qt 4.8.6
PyQt 4.11.3
SIP 4.16.6

![selection_210-edit](https://cloud.githubusercontent.com/assets/9578513/13121420/3e661270-d580-11e5-8b44-5a8f42c574f0.png)

With change you can no longer click on the notification to open the URL.

**Comment by jasonkuehl** [comment]
Notification: title [New message from GUY] message [what-if.xkcd.com/...] icon [/tmp/scudcloud_U08U4Q96C.png]
Notification: title [New message from GUY] message [what-if.xkcd.com/...] icon [/tmp/scudcloud_U08U4Q96C.png]
sys:1: Warning: Source ID 101 was not found when attempting to remove it
URL Changed: https://Place.slack.com/messages/@GUY/

**Comment by raelgc** [comment]
Interesting, as this is printing the pure notification received from Slack... it appears that Slack is cutting the URL.

**Comment by jasonkuehl** [comment]
Yeah This was my thinking as well. The odd thing is if you install the slack client for mac / windows this wont happen. It will open the full URL.

**Comment by raelgc** [comment]
Thanks for all this info, @jasonkuehl.

My guess is: during the init process, ScudCloud pass few configurations to Slack. One of them is to tell Slack that ScudCloud doesn't support HTML notifications (in fact, only because the first version was built with Ubuntu in mind).

So, can you test change this setup?

Please, edit `/usr/lib/python3/dist-packages/scudcloud/resources/scudcloud.js` and change this:

``` javascript
canShowHtmlNotifications:function(){return!1}
```

by this?

``` javascript
canShowHtmlNotifications:function(){return true}
```

Of course, close and re-open ScudCloud and test notifications again.

**Comment by raelgc** [comment]
@jasonkuehl Any lucky?

**Comment by jasonkuehl** [comment]
@raelgc Sorry I’ve been travelling for work. Ill be able to test later on tonight.

**Comment by jasonkuehl** [comment]
Same thing @raelgc 

![selection_214](https://cloud.githubusercontent.com/assets/9578513/13176878/6318d6fc-d6e3-11e5-9b09-6f98a97d2d57.png)

**Comment by stieg** [comment]
Yep @raelgc... did the same locally with top of tree.  Here is the log message: `Notification: title [brent] message [sure -  bit.ly/IqT6zt...] icon [/tmp/scudcloud_U02FC499T.jpg]`.  That is with the suggested change.

**Comment by stieg** [comment]
Ah ha!  progress.  If you pass a function as the value to canShowHtmlNotifications, it treats it as false.  but using a straight boolean yields slack style URLs as defined in their API spec.  So I now pass this: ```canShowHtmlNotifications:true``` and get back URLs in my content that look like this `<google.com>`.  See https://api.slack.com/docs/formatting#linking_to_channels_and_users for some more details...
